manifests/config.pp

```diff
@@ -81,6 +82,10 @@
       order   => '10',
       content => epp("${module_name}/haproxy-base.cfg.epp", $parameters),
     }
+
+    if !empty($programs) {
+      create_resources(haproxy::program, $programs)
```

Comment:
Please don't use `create_resources()`, it's deprecated and we want to get rid of it. please iterate on `$programs` and call the defined resource for each element.

Reply:
@bastelfreak Doesn't seem to be deprecated in [stdlib v9.4.1](https://github.com/puppetlabs/puppetlabs-stdlib/blob/v9.4.1/manifests/manage.pp) where can I find such deprecation note? Generally I'd try to avoid code repetition, this feels like a code smell. I've tried to replace it, but I'm not sure what you mean.